src/screens/BookDetail/index.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useState } from 'react';
+import { useQuery } from 'react-query';
+import { useTranslation } from 'react-i18next';
+import { useHistory, useParams } from 'react-router-dom';
+
+import PATHS from 'constants/paths';
+import { getBookDetail } from 'services/bookService';
+
+import styles from './styles.module.scss';
+
+interface RouteParam {
+  id: string;
+}
+
+interface Details {
+  author: string;
+  editor: string;
+  genre: string;
+  id: number;
+  imageUrl: string;
+  title: string;
+  year: number;
+}
+
+function BookDetail() {
+  const [details, setDetails] = useState<Details>();
+  const { t } = useTranslation();
+  const history = useHistory();
+  const { id } = useParams<RouteParam>();
+  const { error, isLoading } = useQuery('book-detail', () => getBookDetail(id), {
+    onSuccess: (response: Details) => {
+      setDetails(response);
```

Comment:
do not do this please, use data. And you should review more the react query documentation